account_payment_line_cancel/models/account_payment_line.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Compassion CH (http://www.compassion.ch)
+# @author: Marco Monzione <marco.mon@windowslive.com>, Emanuel Cino
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, _
+
+
+class AccountCancelPayment(models.Model):
+    _inherit = 'account.payment.line'
+
+    cancel_reason = fields.Char()
+
+    def cancel_line(self):
+        """
+         This method proceed to call private methods to remove payment lines
+         from payment order
+
+        A cancel reason can be put in the cancel_reason field.
+
+        :return: True
+        """
+        # Retrieve all account_move_line(s) associated with the previous
+        # account_payment_line(s).
+        all_account_move_lines = self.mapped('move_line_id')
+
```

Comment:
Personally I don't like this unneeded extra empty lines that only forces to scroll you more. PEP8 also discourages its use.